pages/checkout/klarnapaynow.vue

```diff
@@ -0,0 +1,132 @@
+<template>
+  <div>
+    <div id="payment-page">
+      <div class="container">
+        <div class="payment-container">
+          <div id="component-container" class="payment"></div>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script setup>
+
+import { ref, onMounted } from 'vue';
+import { AdyenCheckout, Klarna } from '@adyen/adyen-web';
+
+// Reactive references
+const sessionId = ref('');
+const redirectResult = ref('');
+
+// Function to invoke server-side API
+async function sendPostRequest(url, data) {
+  const res = await fetch(url, {
+    method: "POST",
+    body: data ? JSON.stringify(data) : "",
+    headers: {
+      "Content-Type": "application/json",
+    },
+  });
+  return await res.json();
+}
+
+// Function to handle the server response
+async function handleServerResponse(res, component) {
+  if (res.action) {
+    component.handleAction(res.action);
+  } else {
+    switch (res.resultCode) {
+      case "Authorised":
+        window.location.href = "/result/success";
+        break;
+      case "Pending":
+      case "Received":
+        window.location.href = "/result/pending";
+        break;
+      case "Refused":
+        window.location.href = "/result/failed";
+        break;
+      default:
+        window.location.href = "/result/error";
+        break;
+    }
+  }
+}
+
+// Function to create AdyenCheckout instance
+async function createAdyenCheckout(session, clientKey) {
+  return AdyenCheckout(
+    {
+      clientKey,
+      session: session,
+      environment: "test",
+      amount: {
+        value: 10000,
+        currency: 'EUR'
+      },
+      locale: "en_US",
+      countryCode: 'NL',
+      showPayButton: true,
+      onPaymentCompleted: (result, component) => {
+        console.info("onPaymentCompleted");
```

Comment:
Improvement: Maybe it's better to put the business logic here directly instead of checking for the result.Authorised / result.Pending, we now have these separate handlers, what do you think?

Reply:
Yep, created different logic per handler